### ISSUE TITLE
Replace "lambda-beta" into "beta-precision" in the command line options

### ISSUE THREAD

**Issue by tvandera** [ExaScience/smurff#3]
Replace "lambda-beta" into "beta-precision" in the command line options
Also change "lambda-beta-early" into "beta-precision-early" in command line options

**Comment by tvandera** [comment]
Also rename enableLambdaBetaSampling to enableBetaPrecisionSampling

MacauPrior side-information is passed as MatrixConfig. A MatrixConfig has an associated NoiseConfig.
* if NoiseConfigType is `fixed`, set MacauPrior beta_precision to noise model precision and enableBetaPrecisionSampling to false
* if NoiseConfigType is `adaptive`, set MacauPrior beta_precision to noise model precision and enableBetaPrecisionSampling to true
*  if NoiseConfigType is something else, complain

**Comment by motoharu-yano** [comment]
Need to rename python references as well.

**Comment by motoharu-yano** [comment]
I have started working on the comments to this issue and there are couple of questions.
From what I understand the best way to do the assignment of beta_precision and enable_beta_precision_sampling based on noise config is in PriorFactory:
https://github.com/ExaScience/smurff/blob/master/lib/smurff-cpp/SmurffCpp/Priors/PriorFactory.h#L36

However here are the questions:
* Side information is indeed passed as MatrixConfig but in the future we will have multiple side informations #100. Which one to pick? Maybe pick train data noise config instead? Technically all the configs have same values, set in:
https://github.com/ExaScience/smurff/blob/master/lib/smurff-cpp/SmurffCpp/Sessions/CmdSession.cpp#L118
* If we are going to set macau prior beta_precision and enable_beta_precision_sampling values based on noise config, do we still need corresponding command line arguments?
* fixed noise model indeed has precision value. However adaptive noise model does not. it hase sn_init and sn_max values. Which value should I take from noise config in this case so that I can set beta_precision value in macau prior?

**Comment by motoharu-yano** [comment]
Thinking about it a bit more.
We want to set `beta_precision` per `side_info` in macau priors. In this case we will need to take the values from `side_info` noise configs. However at the moment we do not have a way to specify noise config settings per side_info.

**Comment by motoharu-yano** [comment]
After discussion we decided the following:
1. Put more effort in ini files. Command line is lower priority.
2. use direct and tolerance per MacauPriorConfigItem.
3. remove enable_beta_precision from MacauPriorConfig - it is not needed anymore. It is set based on noise type.
4. set beta_precision value in macau priors from precision value of noise config. We currently already have a way to specify noise config per side information in ini file. precision setting will also apply to adaptive noise.
5. remove beta_precision from MacauPriorConfigItem. We dont need it anymore since we set this value from noise config.

**Comment by motoharu-yano** [comment]
Changes first has to be done in master.
I have removed enable_beta_precision from command line, config and from ini file. It is set from noise config.
I have removed beta_precision from command line, config and ini file. It is set from noise config.
I have removed old noise model command line options and added single one that allows to specify noise model type and all the properties. From command line perspective - noise model is specified once and is propagated to train data, aux data and side info. From ini file perspective noise models can be set individually.
Direct and tolerance options are not touched. From command line perspective they are specified once and are propagated to macau priors. From ini file perspective I decided not to touch them. In separate branch called issue-100 we already have these options per MacauPriorConfigItem. There is no reason in introducing these types in master. This can be merged later.
